File: docs/mqtt.js
const mqtt = require('mqtt');

// Upstream (your central broker)
const upstream = mqtt.connect('mqtts://mqtt.brewingremote.com:8883', {
  username: 'bridgeuser',
  password: 'bridgepass',
  // If the runtime trusts public CAs, you usually don’t need ca:
  // ca: fs.readFileSync('isrg-root-x1.pem')
});

// Optional: local broker to mirror into (plain localhost)
const local = mqtt.connect('mqtt://127.0.0.1:1883');

upstream.on('connect', () => {
  console.log('Connected to upstream');
  upstream.subscribe('#', err => {
    if (err) console.error('Sub error', err);
    else console.log('Subscribed to all');
  });
});

upstream.on('message', (topic, payload, packet) => {
  // Mirror message into local broker or process it
  local.publish(topic, payload, { retain: packet.retain, qos: packet.qos });
});

// (Optional) Forward local -> upstream too:
local.on('connect', () => {
  local.subscribe('#');
});
local.on('message', (topic, payload, packet) => {
  upstream.publish(topic, payload, { retain: packet.retain, qos: packet.qos });
});

File: docs/mqtt.py
import paho.mqtt.client as mqtt

UP_HOST = "mqtt.brewingremote.com"
UP_PORT = 8883
USERNAME = "bridgeuser"
PASSWORD = "bridgepass"

def on_connect(client, userdata, flags, rc, properties=None):
    if rc == 0:
        print("Connected upstream")
        client.subscribe("#")
    else:
        print("Connect failed rc=", rc)

def on_message(client, userdata, msg):
    # Here you could publish to another local broker or process
    # For illustration, just log
    print(f"{msg.topic} -> {msg.payload[:60]}")

client = mqtt.Client(client_id="bridge-client", protocol=mqtt.MQTTv311)
client.username_pw_set(USERNAME, PASSWORD)
client.tls_set()  # Uses system CA store; add cafile=... if on minimal system
client.on_connect = on_connect
client.on_message = on_message

client.connect(UP_HOST, UP_PORT, keepalive=30)
client.loop_forever()